### PR TITLE
Implementing an interface adapter for communicating with the query engine. 

### PR DESCRIPTION
**Pull request by jedrzej-grabski**
Here is the implementation of the migration adapter. Its functionality still needs to be extended to include relations and more adapter dataclasses. - Issue #8

### CODE REVIEW

**Comment by mbienkowsk** [comment]
* document some of the stuff that isn't yet (migrationelement, baseadapter, etc)
* should we push the example_table to main? if yes, maybe some designated examples folder since it's not to be treated as source code along with the adapters?

File: migration_engine/adapters/base_adapter.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+from dataclasses import dataclass, field
+from typing import Literal
+
+type DataTypes = Literal[""]
+type Constraints = Literal[""]
+
+
+class BaseAdapter(ABC):
+    type DataTypes = Literal[""]
+    type Constraints = Literal[""]
```
Comment (xDepcio) [inline migration_engine/adapters/base_adapter.py:12]: probably remove redundant code here

File: migration_engine/adapters/sqlite3_adapter.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        name: str
+        data_type: DataTypes
+        constraints: list[Constraints] = field(default_factory=list)
+
+    @dataclass
+    class Table:
+        """Table dataclass for SQLite3Adapter"""
+
+        name: str
+        columns: list[SQLite3Adapter.Column] = field(default_factory=list)
```
Comment (xDepcio) [inline migration_engine/adapters/sqlite3_adapter.py:26]: This code is repated from BaseAdapter. Maybe try to make a generic Column and Table in Base adapter and here, just specify its type arguments.

File: migration_engine/adapters/base_adapter.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        name: str
+        data_type: DataTypes
+        constraints: list[Constraints] = field(default_factory=list)
+
+    @dataclass
+    class Table:
+        name: str
+        columns: list[BaseAdapter.Column] = field(default_factory=list)
+
+    tables: list[Table] = []
```
Comment (xDepcio) [inline migration_engine/adapters/base_adapter.py:23]: Maybe interface shouldn't store any state

**Review by xDepcio — APPROVED** [review]
All changes have been made.

File: migration_engine/adapters/sqlite3_adapter.py
Comment (jedrzej-grabski) [inline migration_engine/adapters/sqlite3_adapter.py:1]: Nice! Smart.

**Pull request merged**